### ISSUE TITLE
RESET_SOFT_AND_HALT as a specifiable behavior for monitor reset

### ISSUE THREAD

**Issue by sfarnswo** [stlink-org/stlink#1319]
- [x] I made serious effort to avoid creating duplicate or nearly similar issue

In order to allow developers to isolate and target your respective issue, please take some time to select the check boxes below and fill out each of the following items appropriate to your specific request.

- [x] Programmer/board type: `STLINK/V3`
- [x] Operating system an version: `Linux`
- [x] **stlink tools version** and/or git commit hash: f7fb0a6caaffdb31611b7f31bc4b5f04a04b7eeb
- [x] stlink commandline tool name: `st-util`
- [x] Target chip (and board, if applicable): `N/A`

I am proposing adding this option to be able to specify a "halt" behavior for the "reset" monitor command.   
  
Using the Cortex-Debug extension on VsCode, and when debugging a target without the NRST pin connected, the default monitor reset attempts and succeeds at performing a soft reset in the absence of NRST. However; early breakpoints (such as those in `main` or `Reset_Handler`) do not trigger. This behavior is corrected when re-compiling the stlink-tools suite with the following modification:

``` diff

diff --git a/src/st-util/gdb-server.c b/src/st-util/gdb-server.c
index 9f9d938..5ad7087 100644
--- a/src/st-util/gdb-server.c
+++ b/src/st-util/gdb-server.c
@@ -1317,7 +1317,7 @@ int32_t serve(stlink_t *sl, st_state_t *st) {
                         reply = strdup("E00");
                     }
 
-                    ret = stlink_reset(sl, RESET_AUTO);
+                    ret = stlink_reset(sl, RESET_SOFT_AND_HALT);
                     if (ret) {
                         DLOG("Rcmd: reset failed with reset\n");
                         reply = strdup("E00");

```
Using the modified solution, the software reset succeeds and early breakpoints are triggered.

**Comment by Nightwalker-87** [comment]
Obviously related to #1198.

**Comment by Nightwalker-87** [comment]
Fixed in commit ae7c9a7dd789603d6c0c8be8696594dd1d2025e0.
